philology/src/main/java/com/jcminarro/philology/transformer/SupportToolbarViewTransformer.kt

```diff
@@ -1,25 +1,33 @@
 package com.jcminarro.philology.transformer
 
-import androidx.appcompat.widget.Toolbar
 import android.util.AttributeSet
 import android.view.View
+import androidx.appcompat.widget.Toolbar
 import com.jcminarro.philology.ViewTransformer
 
 internal object SupportToolbarViewTransformer : ViewTransformer {
     private const val TITLE = "title"
     private const val SUBTITLE = "subtitle"
+    private const val STYLE = "style"
     override fun reword(view: View, attributeSet: AttributeSet): View = view.apply {
         when (this) {
             is Toolbar -> reword(attributeSet)
         }
     }
 
     private fun Toolbar.reword(attributeSet: AttributeSet) {
```

Comment:
![image](https://user-images.githubusercontent.com/1655758/69808718-e3ce6900-11f0-11ea-9800-cf0e03deafc3.png)

The implementation of both toolbars transformers is very similar. Maintaining them both in sync can be quite paintful.
Maybe we can reuse the `reword` extension somehow?

Reply:
@ffelin Good idea. You can come up with a clean-up PR. 👍

Reply:
Here is a small patch 👍 
```
Index: philology/src/main/java/com/jcminarro/philology/transformer/SupportToolbarViewTransformer.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- philology/src/main/java/com/jcminarro/philology/transformer/SupportToolbarViewTransformer.kt	(date 1574947734000)
+++ philology/src/main/java/com/jcminarro/philology/transformer/SupportToolbarViewTransformer.kt	(date 1574948494000)
@@ -6,9 +6,6 @@
 import com.jcminarro.philology.ViewTransformer
 
 internal object SupportToolbarViewTransformer : ViewTransformer {
-    private const val TITLE = "title"
-    private const val SUBTITLE = "subtitle"
-    private const val STYLE = "style"
     override fun reword(view: View, attributeSet: AttributeSet): View = view.apply {
         when (this) {
             is Toolbar -> reword(attributeSet)
@@ -16,18 +13,10 @@
     }
 
     private fun Toolbar.reword(attributeSet: AttributeSet) {
-        attributeSet.forEach { index ->
-            when (attributeSet.getAttributeName(index)) {
-                TITLE -> setTextIfExists(attributeSet, index, this::setTitle)
-                SUBTITLE -> setTextIfExists(attributeSet, index, this::setSubtitle)
-                STYLE -> setTextIfExistsInStyle(
-                    context = context,
-                    attributeSet = attributeSet,
-                    styleIndex = index,
-                    attributesFromStyle = intArrayOf(android.R.attr.title, android.R.attr.subtitle),
-                    setTextResActions = listOf(this::setTitle, this::setSubtitle)
-                )
-            }
-        }
+        context.rewordToolbar(
+            attributeSet,
+            setTitle = this::setTitle,
+            setSubtitle = this::setSubtitle
+        )
     }
 }
Index: philology/src/main/java/com/jcminarro/philology/transformer/ToolbarViewTransformer.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- philology/src/main/java/com/jcminarro/philology/transformer/ToolbarViewTransformer.kt	(date 1574947734000)
+++ philology/src/main/java/com/jcminarro/philology/transformer/ToolbarViewTransformer.kt	(date 1574948494000)
@@ -1,16 +1,19 @@
 package com.jcminarro.philology.transformer
 
+import android.R.attr
 import android.annotation.SuppressLint
+import android.content.Context
 import android.util.AttributeSet
 import android.view.View
 import android.widget.Toolbar
 import com.jcminarro.philology.ViewTransformer
 
-@SuppressLint("NewApi")
-internal object ToolbarViewTransformer : ViewTransformer {
-    private const val TITLE = "title"
-    private const val SUBTITLE = "subtitle"
-    private const val STYLE = "style"
+private const val TITLE = "title"
+private const val SUBTITLE = "subtitle"
+private const val STYLE = "style"
+
+@SuppressLint("NewApi")
+internal object ToolbarViewTransformer : ViewTransformer {
     override fun reword(view: View, attributeSet: AttributeSet): View = view.apply {
         when (this) {
             is Toolbar -> reword(attributeSet)
@@ -18,18 +21,38 @@
     }
 
     private fun Toolbar.reword(attributeSet: AttributeSet) {
-        attributeSet.forEach { index ->
-            when (attributeSet.getAttributeName(index)) {
-                TITLE -> setTextIfExists(attributeSet, index, this::setTitle)
-                SUBTITLE -> setTextIfExists(attributeSet, index, this::setSubtitle)
-                STYLE -> setTextIfExistsInStyle(
-                    context = context,
-                    attributeSet = attributeSet,
-                    styleIndex = index,
-                    attributesFromStyle = intArrayOf(android.R.attr.title, android.R.attr.subtitle),
-                    setTextResActions = listOf(this::setTitle, this::setSubtitle)
-                )
-            }
+        context.rewordToolbar(
+            attributeSet,
+            setTitle = this::setTitle,
+            setSubtitle = this::setSubtitle
+        )
+    }
+}
+
+fun Context.rewordToolbar(
+    attributeSet: AttributeSet,
+    setTitle: (Int) -> Unit,
+    setSubtitle: (Int) -> Unit
+) {
+    attributeSet.forEach { index ->
+        when (attributeSet.getAttributeName(index)) {
+            TITLE -> ToolbarViewTransformer.setTextIfExists(
+                attributeSet,
+                index,
+                setTitle
+            )
+            SUBTITLE -> ToolbarViewTransformer.setTextIfExists(
+                attributeSet,
+                index,
+                setSubtitle
+            )
+            STYLE -> ToolbarViewTransformer.setTextIfExistsInStyle(
+                context = this,
+                attributeSet = attributeSet,
+                styleIndex = index,
+                attributesFromStyle = intArrayOf(attr.title, attr.subtitle),
+                setTextResActions = listOf(setTitle, setSubtitle)
+            )
         }
     }
 }
```

Reply:
@ffelini I have added unit tests and know this patch fails them. Will leave it for now.